app/controllers/api/notes_controller.rb

```diff
@@ -401,6 +401,8 @@ def add_comment(note, text, event, notify: true)
       note.comments.map(&:author).uniq.each do |user|
         UserMailer.note_comment_notification(comment, user).deliver_later if notify && user && user != current_user && user.visible?
       end
+
+      NoteSubscription.find_or_create_by(:note => note, :user => current_user) if current_user
```

Comment:
I think this could be simplified using the association?

```suggestion
      current_user&.note__subscriptions.find_or_create_by(:note => note)
```

Reply:
That would require one more `&.` after `note__subscriptions`, so it depends on whether you think that chained `&.`s are simplifications.

Reply:
I don't believe it does - if `current_user` is defined then the `note_subscriptions` association will always exist. It might evaluate to an empty list but the association object will exist and can have methods invoked on it.

Reply:
Try running Rubocop.

Reply:
Hmm you're quite right, and rubocop is quite right, which I find quite odd but apparently ruby does continue evaluating the line after the first `&.` fails.

Personally I still think I prefer it but it's not a huge issue.